scripts/cp-pkgs.js

```diff
@@ -68,18 +88,92 @@ function getAssetEntry(asset) {
   return [MDC, toSlugCase(componentName)].join('-');
 }
 
-function cpAsset(asset) {
+/**
+ * @param {string} asset filepath relative to the root directory
+ * @returns {Promise<void>}
+ */
+async function cpAsset(asset) {
   const assetPkg = path.join('packages', getAssetEntry(asset));
   if (!fs.existsSync(assetPkg)) {
-    Promise.reject(new Error(`Non-existent asset package path ${assetPkg} for ${asset}`));
+    return Promise.reject(new Error(`Non-existent asset package path ${assetPkg} for ${asset}`));
   }
   const destDir = path.join(assetPkg, 'dist', path.basename(asset));
-  return cpFile(asset, destDir).then(() => console.log(`cp ${asset} -> ${destDir}`));
+  return cpFile(asset, destDir).then(
+    () => console.log(`cp ${asset} -> ${destDir}`),
+    (err) => {
+      throw err;
+    }
+  );
+}
+
+/**
+ * Imports all files in index.d.ts and compiles a bundled .d.ts file for
+ * UMD files.
+ */
+function dtsBundler() {
+  const packageDirectories = fs.readdirSync(D_TS_DIRECTORY);
+  packageDirectories.forEach((packageDirectory) => {
+    const main = path.resolve(D_TS_DIRECTORY, packageDirectory, './index.d.ts');
+    dts.bundle({
+      name: packageDirectory,
+      main,
+    });
+  });
+}
+
+/**
+ * @param {string} asset filepath relative to the root directory
+ * @param {string} assetPkg directory path to destination package
+ * @returns {string} destination filepath for UMD declaration file.
+ */
+function getDeclarationFileName(asset, assetPkg) {
+  const packageName = path.parse(asset).name.replace(/^mdc-|\.d$/g, '');
+  const isAllInOne = packageName === ALL_IN_ONE_PACKAGE;
+  const destFileName = isAllInOne ? packageName : `mdc.${toCamelCase(packageName)}`;
+  return path.join(assetPkg, 'dist', `${destFileName}.d.ts`);
 }
 
+/**
+ * Copies the declaration file from the /build directory to its respective
+ * destination directory.
+ * @param {string} asset filepath relative to the root directory
+ * @returns {Promise<void>}
+ */
+async function cpDeclarationAsset(asset) {
+  const assetPkg = path.parse(asset.split('build/')[1]).dir;
+  if (!fs.existsSync(assetPkg)) {
+    return Promise.reject(new Error(`Non-existent asset package path ${assetPkg} for ${asset}`));
+  }
+
+  const destDir = getDeclarationFileName(asset, assetPkg);
+  return cpFile(asset, destDir).then(
+    () => console.log(`cp ${asset} -> ${destDir}`),
+    (err) => {
+      throw err;
+    }
+  );
+}
+
+/**
```

Comment:
Nit: Remove second `*` here and add a newline after `*/`. This comment describes everything below it; it's not a JSDoc comment for `cleanPkgDistDirs()` specifically.